Is the use of _pkgdown.yml file now mandatory

Hi,

I am using `pkgdown` with Github actions to build elegant sites for my R packages for some months now.
Since last week, however, my CI/CD process returns an error message, complaining that no `_pkgdown.yml` file can be found.
I had never used a `_pkgdown.yml` file before (no customization) in my R packages. 
Is this now a requirement of the latest `pkgdown` version ? Or a regression ? 

Here below is a log with the error message and Backtrace I am getting. 

Thanks, Philippe. 

```
2024-05-14T08:48:04.9207226Z ##[group]Run git config --global user.email "actions@github.com"
2024-05-14T08:48:04.9207807Z [36;1mgit config --global user.email "actions@github.com"[0m
2024-05-14T08:48:04.9208596Z [36;1mgit config --global user.name "GitHub Actions"[0m
2024-05-14T08:48:04.9209196Z [36;1mgit config --global --add safe.directory /__w/CytoPipelineGUI/CytoPipelineGUI[0m
2024-05-14T08:48:04.9209840Z [36;1mRscript -e "pkgdown::deploy_to_branch(new_process = FALSE)"[0m
2024-05-14T08:48:04.9210357Z shell: bash {0}
2024-05-14T08:48:04.9210583Z env:
2024-05-14T08:48:04.9210790Z   has_testthat: true
2024-05-14T08:48:04.9211044Z   run_covr: true
2024-05-14T08:48:04.9211277Z   run_pkgdown: true
2024-05-14T08:48:04.9211517Z   has_RUnit: false
2024-05-14T08:48:04.9211752Z   has_BiocCheck: true
2024-05-14T08:48:04.9212011Z   cache-version: cache-v1
2024-05-14T08:48:04.9212309Z   R_REMOTES_NO_ERRORS_FROM_WARNINGS: true
2024-05-14T08:48:04.9212798Z   RSPM: https://packagemanager.rstudio.com/cran/__linux__/focal/latest
2024-05-14T08:48:04.9213282Z   NOT_CRAN: true
2024-05-14T08:48:04.9213508Z   TZ: UTC
2024-05-14T08:48:04.9213728Z   VDIFFR_RUN_TESTS: false
2024-05-14T08:48:04.9214141Z   GITHUB_TOKEN: ***
2024-05-14T08:48:04.9214490Z   GITHUB_PAT: ***
2024-05-14T08:48:04.9214796Z   CODECOV_TOKEN: 409bdac0-11db-44df-b4b2-b52d85a3ab92
2024-05-14T08:48:04.9215159Z ##[endgroup]
2024-05-14T08:48:05.5048066Z Running git remote set-branches --add origin gh-pages
2024-05-14T08:48:05.5114683Z Running git fetch origin gh-pages
2024-05-14T08:48:06.2252897Z From https://github.com/UCLouvain-CBIO/CytoPipelineGUI
2024-05-14T08:48:06.2253836Z  * branch            gh-pages   -> FETCH_HEAD
2024-05-14T08:48:06.2254442Z  * [new branch]      gh-pages   -> origin/gh-pages
2024-05-14T08:48:06.2569006Z Adding worktree
2024-05-14T08:48:06.2579162Z Running git worktree add --track -B gh-pages /tmp/Rtmp3J5hDB/filea285a9d13c8 \
2024-05-14T08:48:06.2580001Z   origin/gh-pages
2024-05-14T08:48:06.2630983Z Preparing worktree (new branch 'gh-pages')
2024-05-14T08:48:06.2657888Z Branch 'gh-pages' set up to track remote branch 'gh-pages' from 'origin'.
2024-05-14T08:48:06.2889406Z HEAD is now at 8fb5d21 cli-147-285
2024-05-14T08:48:06.6502097Z ── Cleaning files from old site ────────────────────────────────────────────────
2024-05-14T08:48:06.6671070Z Cleaning CytoPipelineGUI pkgdown docs from '/tmp/Rtmp3J5hDB/filea285a9d13c8'
2024-05-14T08:48:06.6946329Z ── Building pkgdown site for package CytoPipelineGUI ───────────────────────────
2024-05-14T08:48:06.7007705Z Reading from: /__w/CytoPipelineGUI/CytoPipelineGUI
2024-05-14T08:48:06.7066464Z Writing to: /tmp/Rtmp3J5hDB/filea285a9d13c8
2024-05-14T08:48:06.7081904Z ── Sitrep ──────────────────────────────────────────────────────────────────────
2024-05-14T08:48:07.4799860Z Error in `fun(..., .envir = .envir)`:
2024-05-14T08:48:07.4800783Z ! Could not evaluate cli `{}` expression: `config_path(pkg)`.
2024-05-14T08:48:07.4802641Z Caused by error in `config_path(pkg)`:
2024-05-14T08:48:07.4803547Z ! Can't find '_pkgdown.yml'.
2024-05-14T08:48:07.4804662Z ℹ This is an internal error that was detected in the pkgdown package.
2024-05-14T08:48:07.4806363Z   Please report it at <https://github.com/r-lib/pkgdown/issues> with a reprex (<https://tidyverse.org/help/>) and the full backtrace.
2024-05-14T08:48:07.4807847Z ---
2024-05-14T08:48:07.4808394Z Backtrace:
2024-05-14T08:48:07.4808998Z  1. pkgdown::deploy_to_branch(new_process = FALSE)
2024-05-14T08:48:07.4809875Z  2. pkgdown::build_site_github_pages(pkg, ..., clean = clean)
2024-05-14T08:48:07.4810964Z  3. pkgdown::build_site(pkg, preview = FALSE, install = install, new_process = new_process, …
2024-05-14T08:48:07.4812295Z  4. pkgdown:::build_site_local(pkg = pkg, examples = examples, run_dont_run = run_dont_run, …
2024-05-14T08:48:07.4813247Z  5. pkgdown::pkgdown_sitrep(pkg)
2024-05-14T08:48:07.4813895Z  6. pkgdown:::error_to_sitrep("URLs", check_urls(pkg))
2024-05-14T08:48:07.4814705Z  7. base::tryCatch({ …
2024-05-14T08:48:07.4815330Z  8. base::tryCatchList(expr, classes, parentenv, handlers)
2024-05-14T08:48:07.4816781Z  9. base::tryCatchOne(expr, names, parentenv, handlers[[1L]])
2024-05-14T08:48:07.4817704Z 10. base::doTryCatch(return(expr), name, parentenv, handler)
2024-05-14T08:48:07.4818456Z 11. pkgdown:::check_urls(pkg)
2024-05-14T08:48:07.4819226Z 12. cli::cli_abort(c("{config_path(pkg)} lacks {.field url}.", details), …
2024-05-14T08:48:07.4820198Z 13. cli:::vcapply(message, format_inline, .envir = .envir)
2024-05-14T08:48:07.4821205Z 14. base::vapply(X, FUN, FUN.VALUE = character(1), ..., USE.NAMES = USE.NAMES)
2024-05-14T08:48:07.4822059Z 15. local FUN(X[[i]], ...)
2024-05-14T08:48:07.4822864Z 16. cli::cli_fmt(fun(..., .envir = .envir), collapse = collapse, strip_newline = TRUE)
2024-05-14T08:48:07.4823726Z 17. cli:::cli__rec(expr)
2024-05-14T08:48:07.4824214Z 18. local fun(..., .envir = .envir)
2024-05-14T08:48:07.4825050Z 19. cli:::cli__message("inline_text", list(text = glue_cmd(..., .envir = .envir, …
2024-05-14T08:48:07.4825889Z 20. "id" %in% names(args)
2024-05-14T08:48:07.4826596Z 21. cli:::glue_cmd(..., .envir = .envir, .call = sys.call(), .trim = FALSE)
2024-05-14T08:48:07.4827644Z 22. cli:::glue(str, .envir = .envir, .transformer = transformer, .cli = TRUE, …
2024-05-14T08:48:07.4828552Z 23. (function (expr) …
2024-05-14T08:48:07.4829084Z 24. .transformer(expr, .envir) %||% character()
2024-05-14T08:48:07.4829715Z 25. local .transformer(expr, .envir)
2024-05-14T08:48:07.4830961Z 26. eval(expr, envir = envir) %??% cli_error(call. = caller, "Could not evaluate cli {.code {{}}} expression:\n           {.code {abbrev(code, 20)}}.")
2024-05-14T08:48:07.4832433Z 27. cli:::chain_error(expr, err, srcref = utils::getSrcref(sys.call()))
2024-05-14T08:48:07.4833249Z 28. | base::withCallingHandlers({ …
2024-05-14T08:48:07.4833811Z 29. base::eval(expr, envir = envir)
2024-05-14T08:48:07.4834372Z 30. base::eval(expr, envir = envir)
2024-05-14T08:48:07.4834905Z 31. pkgdown:::config_path(pkg)
2024-05-14T08:48:07.4835737Z 32. cli::cli_abort("Can't find {.file _pkgdown.yml}.", .internal = TRUE)
2024-05-14T08:48:07.4836745Z 33. | rlang::abort(message, ..., call = call, use_cli_format = TRUE, …
2024-05-14T08:48:07.4837547Z 34. | rlang:::signal_abort(cnd, .file)
2024-05-14T08:48:07.4838119Z 35. | base::signalCondition(cnd)
2024-05-14T08:48:07.4838678Z 36. | (function (e) …
2024-05-14T08:48:07.4839149Z 37. | cli:::throw_error(err, parent = e)
2024-05-14T08:48:07.4869149Z Removing worktree
2024-05-14T08:48:07.4877966Z Running git worktree remove /tmp/Rtmp3J5hDB/filea285a9d13c8
2024-05-14T08:48:07.4959101Z Execution halted
2024-05-14T08:48:07.5096234Z ##[error]Process completed with exit code 1.
```

--- a/R/config.R
+++ b/R/config.R
@@ -128,9 +128,7 @@ config_abort <- function(pkg,
 }
 
 config_path <- function(pkg) {
-  config <- pkgdown_config_path(pkg$src_path)
-  if (is.null(config)) {
-    cli::cli_abort("Can't find {.file _pkgdown.yml}.", .internal = TRUE)
-  }
-  cli::style_hyperlink(path_file(config), paste0("file://", config))
+  # Not all projects necessary have a _pkgdown.yml (#2542)
+  config <- pkgdown_config_path(pkg$src_path) %||% "_pkgdown.yml"
+  cli::style_hyperlink(path_file(config), paste0("file://", config))  
 }

--- a/R/pkgdown.R
+++ b/R/pkgdown.R
@@ -38,8 +38,6 @@ local_pkgdown_site <- function(path = NULL, meta = NULL, env = parent.frame()) {
     desc$set("Package", "testpackage")
     desc$set("Title", "A test package")
     desc$write(file = path(path, "DESCRIPTION"))
-    
-    file_create(path(path, "_pkgdown.yml"))
   }
 
   if (is.character(meta)) {

